db/columns/utils.py

```diff
@@ -41,3 +41,24 @@ def init_mathesar_table_column_list_with_defaults(column_list):
             if c.name == constants.ID:
                 c.autoincrement = False
     return default_columns + given_columns
+
+
+def get_column_obj_from_relation(relation, column):
+    try:
+        column = find_column_by_name_in_relation(relation, column)
+    except AttributeError:
+        column = relation.columns[column.name]
+
+    return column
+
+
+# TODO deal with quotes; still better than the default
+def find_column_by_name_in_relation(relation, col_name_string):
```

Comment:
Hm, seems a bit weird that we have two methods looking up column names, one delegating to the other, but then also both primarily attempting to do column indexing (`relation.columns[...]`). I'm having trouble understanding the intent and the challenges faced here.

Reply:
The outer function finds a column in the relation whether it's a `Column`, `MathesarColumn`, or just a name string. It first delegates finding by string to the inner function, since that needs to handle different levels of namespacing (could be schema.table.name, table.name, or just name) and it's a bit complicated. The inner function will return an `AttributeError` when it's not passed a string (or something string-like enough to have a `split` method). In that case, the outer function tries to just get the table-associated version of column by looking for one with the name the same as whatever object it's been passed.